packages/mgt-components/src/graph/graph.files.ts

```diff
@@ -573,6 +1064,51 @@ async function getDriveItemsByInsights(graph: IGraph, insightResponse, scopes) {
   }
 }
 
-async function getFilesPageIterator(graph: IGraph, request) {
+async function getFilesPageIteratorFromRequest(graph: IGraph, request) {
   return GraphPageIterator.create<DriveItem>(graph, request);
 }
+
+async function getFilesPageIteratorFromCache(graph: IGraph, value, nextLink) {
+  return GraphPageIterator.createFromValue<DriveItem>(graph, value, nextLink);
+}
+
+async function getFileFromCache(cache: CacheStore<CacheFile>, key: string) {
+  if (getIsFilesCacheEnabled()) {
+    const file = await cache.getValue(key);
+
+    if (file && getFileInvalidationTime() > Date.now() - file.timeCached) {
+      const cachedFile = JSON.parse(file.file);
+      return cachedFile;
+    }
+  }
+}
+
+export async function getFileListFromCache(cache: CacheStore<CacheFileList>, key: string) {
+  if (!cache) {
+    cache = CacheService.getCache<CacheFileList>(schemas.fileLists, schemas.fileLists.stores.fileLists);
+  }
+
+  if (getIsFileListsCacheEnabled()) {
+    const fileList = await cache.getValue(key);
+
+    if (fileList && getFileListInvalidationTime() > Date.now() - fileList.timeCached) {
+      return fileList;
+    }
+  }
+}
+
+// refresh filesPageIterator to its next iteration and save current page to cache
+export async function getNextFilesPageIterator(filesPageIterator) {
```

Comment:
Consider renaming this function - I was expecting it to return something and took me a bit to understand what it does. Something along the lines of `fetchNextAndCacheForFilesPageIterator` might be more descriptive.